src/common/src/array/stream_chunk.rs

```diff
@@ -151,6 +152,16 @@ impl StreamChunk {
         &self.data
     }
 
+    pub fn ends_with_update(&self) -> bool {
```

Comment:
I thought this is a somehow general property of a chunk. No preference personally. Since only rate limit use it for now, I'll move this to rate limit part.